src/node.rs

```diff
@@ -466,11 +466,19 @@ impl PbftNode {
         &mut self,
         state: &mut PbftState,
         block: &Block,
-        msg: PbftMessage,
+        msg: &PbftMessage,
+        head: &Block,
     ) -> Result<bool, PbftError> {
-        // Don't catch up if we're the primary, since we're the ones who publish.
-        // Also don't catch up from block 0 -> 1, since there's no consensus seal.
-        if state.is_primary() || block.block_num < 2 {
+        info!(
+            "{}: Trying catchup from head #{} to #{}, from BlockNew message #{}",
+            state,
+            head.block_num,
+            head.block_num + 1,
+            block.block_num,
+        );
+
+        // Don't catch up from block 0 -> 1, since there's no consensus seal.
+        if block.block_num < 2 {
```

Comment:
Not returning early when the node is primary seems like a pretty significant change that isn't really captured by the commit message; this should probably be in its own commit.